olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp

```diff
@@ -207,24 +290,70 @@ TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyNotFound) {
       .WillOnce(testing::Return(boost::any()));
 
   EXPECT_CALL(*network_, Send(IsGetRequest(kLookupMetadata), _, _, _, _))
-      .Times(1)
       .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
                                        olp::http::HttpStatusCode::NOT_FOUND),
                                    ""));
 
   ApiLookupClient lookup_client(kHrn, settings_);
   repository::CatalogRepository repository(kHrn, settings_, lookup_client);
-  auto response = repository.GetLatestVersion(request, context);
 
-  EXPECT_FALSE(response.IsSuccessful());
+  std::promise<read::CatalogVersionResponse> promise;
+  auto future = promise.get_future();
+  const auto response = repository.GetLatestVersion(
+      request, [&](read::CatalogVersionResponse response) {
+        promise.set_value(std::move(response));
+      });
+
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  const auto result = future.get();
+
+  EXPECT_FALSE(result);
+  EXPECT_EQ(result.GetError().GetErrorCode(), olp::client::ErrorCode::NotFound);
 }
 
-TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyFound) {
-  olp::client::CancellationContext context;
+TEST_F(CatalogRepositoryTest, AsyncGetLatestVersionOnlineOnlyForbidden) {
+  auto request =
+      read::CatalogVersionRequest().WithFetchOption(read::OnlineIfNotFound);
 
-  auto request = read::CatalogVersionRequest();
+  ON_CALL(*cache_, Get(_, _))
+      .WillByDefault([](const std::string&, const olp::cache::Decoder&) {
+        ADD_FAILURE() << "Cache should not be used in OnlineOnly request";
```

Comment:
But you have `OnlineIfNotFound` request here...
Do you really need this default action, or it is leftover from some copy-paste?